### DISCUSSION TITLE
Completely Silent Install

### DISCUSSION THREAD

**Discussion by Anti-Distinctlyminty** [microsoft/winget-cli#1578] [Q&A] [answered]
Hello,
I've just started looking into winget as a tool for deploying software, and during my testing I noticed that quite often software installs or uninstalls require user intervention to click a button or two. What I need is a _completly_ silent install that will require no user intervention whatsoever.
Is this behaviour that I'm seeing a result of winget itself or the packages themselves being incorrectly set up?

**Comment by denelon** [comment, marked as answer]
@Anti-Distinctlyminty the packages themselves need to support a silent uninstall experience in order for the Windows Package Manager to perform the uninstall silently. In some cases there may be switches or flags required. We're looking at the options for this behavior. It may require adding a switches section for uninstall in the manifest schema.

**Reply by Anti-Distinctlyminty** [reply]
> the packages themselves need to support a silent uninstall experience...

Just to clarify, by "packages" here you mean the msi or exe? I've had success with doing silent installs using wpkg and chocolatey, so if that's the case and we are able to host our own package repository, then this is most promising.

**Reply by denelon** [reply]
Yes 😊
